src/app/clusters/general-commissioning-server/general-commissioning-server.cpp

```diff
@@ -105,52 +103,47 @@ CHIP_ERROR GeneralCommissioningAttrAccess::Read(const ConcreteReadAttributePath
     }
 #if CHIP_CONFIG_TC_REQUIRED
     case TCAcceptedVersion::Id: {
-        TermsAndConditionsProvider * const termsAndConditionsProvider = TermsAndConditionsManager::GetInstance();
+        TermsAndConditionsProvider * tcProvider = TermsAndConditionsManager::GetInstance();
```

Comment:
This could be injected in the General Commissioning GeneralCommissioningAttrAccess, rather than using a singleton. This can be revisited in a follow-up, no need to change now.

Please avoid all singletons as they make code harder to scale and test, and prevent hosting multiple separate nodes in the same process if there is a data dependency on the singleton. Every new singleton is tech debt.

Reply:
Created issue to track this: https://github.com/project-chip/connectedhomeip/issues/36832